src/definitions.jl

```diff
@@ -561,3 +576,58 @@ Pre-plan an optimized real-input unnormalized transform, similar to
 the same as for [`brfft`](@ref).
 """
 plan_brfft
+
+##############################################################################
+
+region(p::Plan) = p.region
+region(p::ScaledPlan) = region(p.p)
+
+# Projection style (:none, :real, or :real_inv) to handle real FFTs
+function projection_style end
+# Length of halved dimension, needed only for irfft 
+function irfft_dim end
+
+mutable struct AdjointPlan{T,P} <: Plan{T}
+    p::P
+    pinv::Plan
+    AdjointPlan{T,P}(p) where {T,P} = new(p)
+    # always have adjoint inside scaled
+    AdjointPlan{T,P}(p::P) where {T,P<:ScaledPlan{T}} = ScaledPlan{T}(AdjointPlan{T}(p.p), p.scale)
+    AdjointPlan{T,P}(p::AdjointPlan{T}) where {T,P} = new(p.p)
+end
+
+AdjointPlan{T}(p::P) where {T,P} = AdjointPlan{T,P}(p)
+AdjointPlan(p::Plan{T}) where {T} = AdjointPlan{T}(p)
+Base.adjoint(p::Plan{T}) where {T} = AdjointPlan{T}(p)
+
+size(p::AdjointPlan) = output_size(p)
+output_size(p::AdjointPlan) = size(p)
+
+function Base.:*(p::AdjointPlan{T}, x::AbstractArray) where {T}
+    dims = region(p.p)
+    halfdim = first(dims)
+    d = size(p.p, halfdim)
+    n = output_size(p.p, halfdim)
+    if projection_style(p.p) == :none
+        N = normalization(T, size(p.p), dims)
+        return 1/N * (p.p \ x)
+    elseif projection_style(p.p) == :real
+        N = normalization(T, size(p.p), dims)
+        scale = reshape(
+            [(i == 1 || (i == n && 2 * (i - 1)) == d) ? 1 : 2 for i in 1:n],
+            ntuple(i -> i == first(dims) ? n : 1, Val(ndims(x)))
+        )
+        return 1/N * (p.p \ (x ./ scale))
+    elseif projection_style(p.p) == :real_inv
+        N = normalization(real(T), output_size(p.p), dims)
+        scale = reshape(
+            [(i == 1 || (i == d && 2 * (i - 1)) == n) ? 1 : 2 for i in 1:d],
+            ntuple(i -> i == first(dims) ? d : 1, Val(ndims(x)))
+        )
+        return 1/N * scale .* (p.p \ x)
+    else
+        error("plan must define a valid projection style")
```

Comment:
Not covered by the tests. Again maybe it would be easier to just dispatch on `AdjointPlan` directly?